WalletWasabi.Gui/Global.cs

```diff
@@ -285,9 +281,11 @@ private void RegisterFeeRateProviders()
 			HostedServices.Register<HybridFeeProvider>(new HybridFeeProvider(HostedServices.Get<ThirdPartyFeeProvider>(), HostedServices.GetOrDefault<RpcFeeProvider>()), "Hybrid Fee Provider");
 		}
 
-		private void RegisterCoinJoinComponents()
+		private void RegisterCoinJoinComponents(SystemAwakeChecker systemAwakeChecker)
 		{
 			HostedServices.Register<RoundStateUpdater>(new RoundStateUpdater(TimeSpan.FromSeconds(5), new WabiSabiHttpApiClient(BackendHttpClientFactory.NewBackendHttpClient(Mode.SingleCircuitPerLifetime))), "Round infor updater");
+
+			// Send systemAwakeChecker into CoinJoinManager via ctor
```

Comment:
Please no. An event or a callback should be enough. Coupling of `CoinJoinManager` with `SystemAwakeChecker` is not necessary and it hurts our ability to test the code.

(It would be great to have a unit test here.)

Reply:
Yes, that needs to change.